llvm/include/llvm/IR/Instructions.h

```diff
@@ -4942,6 +4942,25 @@ inline std::optional<SyncScope::ID> getAtomicSyncScopeID(const Instruction *I) {
   llvm_unreachable("unhandled atomic operation");
 }
 
+/// A helper function that sets an atomic operation's sync scope.
+/// Does nothing if it is not an atomic operation.
+inline void setAtomicSyncScopeID(Instruction *I, SyncScope::ID SSID) {
+  if (!I->isAtomic())
```

Comment:
I think I'd prefer an `assert(I->isAtomic())` here. Silently discarding a set seems like a bad idea. The caller can do an isAtomic() check if it wants that behavior.